src/target/stm32mp15.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * This file is part of the Black Magic Debug project.
+ *
+ * Copyright (C) 2023 1BitSquared <info@1bitsquared.com>
+ * Written by ALTracer <tolstov_den@mail.ru>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * This file implements STM32MP15 target specific functions for detecting
+ * the device and providing the XML memory map.
+ *
+ * References:
+ * RM0436 - STM32MP157 advanced Arm®-based 32-bit MPUs, Rev. 5
+ *   https://www.st.com/resource/en/reference_manual/rm0436-stm32mp157-advanced-armbased-32bit-mpus-stmicroelectronics.pdf
+ */
+
+#include "general.h"
+#include "target.h"
+#include "target_internal.h"
+#include "cortexm.h"
+
+/* Access from processor address space.
+ * Access via the debug APB is at 0xe0081000 over AP1. */
+#define DBGMCU_IDCODE      0x50081000U
```

Comment:
Same comment for all the DBG* macros here as in #1547 - pleas prefix with the MCU series name like you did on the next macro down.

Additionally, as you're using this as a base address, it should be `STM32MP15_DBGMCU_BASE` and the macro you've named _IDC, should be `_IDCODE`.

Reply:
Prefixed as suggested, for the `STM32MP15_DBGMCU_BASE` and `STM32MP15_UID_BASE`
Not touching the _IDC and _CR (yet) because that's what these registers were always called across most of STM32 lines of MCUs, and the layout seems to match across stm32** BMP target drivers.

Reply:
This naming convention is something we've been working on fixing. Changing this to match the convention being established reduces the number of fixups we have to do later.

Reply:
This item still needs applying to lines 40 through 44, along with the register naming convention point in the next review comment down